### PR TITLE
[devp2p] Update to 2018 edition

### PR DESCRIPTION
**Pull request by dvdplm**
Builds on https://github.com/paritytech/parity-ethereum/pull/10714

Enables NLL on devp2p

### CODE REVIEW

File: util/network-devp2p/src/connection.rs [hunk 1]
```diff
@@ -317,14 +318,13 @@ impl EncryptedConnection {
...
-		let decoder = CtrMode::new(AesSafe256Encryptor::new(&key_material[32..64]), iv);
-
-		let key_material_keccak = keccak(&key_material);
-		(&mut key_material[32..64]).copy_from_slice(key_material_keccak.as_bytes());
-		let mac_encoder = EcbEncryptor::new(AesSafe256Encryptor::new(&key_material[32..64]), NoPadding);
+		// TODO: clarify this: ecdh::agree creates a **NEW** secret right? And AesCtr256 keeps an internal counter, right?
+		// Using a 0 IV with CTR is fine as long as the same IV is never reused with the same key. This is not the case here.
+		let encoder = AesCtr256::new(&key_material[32..64], &NULL_IV)?;
+		let decoder = AesCtr256::new(&key_material[32..64], &NULL_IV)?;
+        let key_material_keccak = keccak(&key_material);
```
Comment (niklasad1) [inline util/network-devp2p/src/connection.rs]: indentation

File: util/network-devp2p/src/connection.rs [hunk 2]
```diff
@@ -317,14 +318,13 @@ impl EncryptedConnection {
...
+        (&mut key_material[32..64]).copy_from_slice(key_material_keccak.as_bytes());
```
Comment (niklasad1) [inline util/network-devp2p/src/connection.rs]: indentation

**Comment by niklasad1** [comment]
@dvdplm it is very hard to review, can you merge it into master? Then I think all changes from #10714 will be gone

**Comment by dvdplm** [comment]
@niklasad1 apologies, slightly better now. Thanks!

File: util/network-devp2p/src/service.rs [hunk 3]
```diff
@@ -18,14 +18,19 @@ use std::net::SocketAddr;
 use std::ops::RangeInclusive;
 use std::sync::Arc;
 
+//TODO: remove this
```
Comment (ordian) [inline util/network-devp2p/src/service.rs]: > TODOs should be tickets

Do we have a ticket for that :P

Comment (dvdplm) [reply util/network-devp2p/src/service.rs]: I didn't add the `TODO` but it illustrates exactly my point: we're carrying around a todo-comment nobody even knows what it means anymore. The information content is tending towards zero...
I'll actually just remove it.

File: util/network-devp2p/src/session.rs [hunk 4]
```diff
@@ -372,7 +376,7 @@ impl Session {
 			},
 			PACKET_GET_PEERS => Ok(SessionData::None), //TODO;
 			PACKET_PEERS => Ok(SessionData::None),
-			PACKET_USER ... PACKET_LAST => {
+			PACKET_USER ..= PACKET_LAST => {
```
Comment (niklasad1) [inline util/network-devp2p/src/session.rs]: `range_pattern` is not deprecated on stable yet right or have I missed something?

However, good catch all good from my side

**Pull request merged**